File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.*;

class Solution {
    public int solution(int[][] tri) {
        for (int i = 1; i < tri.length; i++) {
            for (int j = 0; j < tri[i].length; j++) {
                if (j == 0) {
                    tri[i][j] += tri[i-1][0];
                } else if (j == tri[i].length - 1) {
                    tri[i][j] += tri[i-1][j-1];
                } else {
                    tri[i][j] += Math.max(tri[i-1][j-1], tri[i-1][j]);
                }
            }
        }
        
        List<Integer> answer = new ArrayList<>();
        for (int num: tri[tri.length - 1]) {
            answer.add(num);
        }            
        return Collections.max(answer);
    }
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(tri):
    
    for i in range(1, len(tri)):
        for j in range(len(tri[i])):
            if j == 0:
                tri[i][j] += tri[i-1][0]
            elif j == len(tri[i-1]):
                tri[i][j] += tri[i-1][j-1] 
            else:
                tri[i][j] += max(tri[i-1][j-1], tri[i-1][j]) 

    return max(tri[-1])